registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/utils/NotificationUtility.java

```diff
@@ -136,20 +152,64 @@ public void sendNotification(RegistrationAdditionalInfoDTO registrationAdditiona
 				if (notificationType.equalsIgnoreCase("EMAIL")
 						&& (registrationAdditionalInfoDTO.getEmail() != null
 						&& !registrationAdditionalInfoDTO.getEmail().isEmpty())) {
-					sendEmailNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description);
+					sendEmailNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description,preferredLanguage);
 				} else if (notificationType.equalsIgnoreCase("SMS") && (registrationAdditionalInfoDTO.getPhone() != null
 						&& !registrationAdditionalInfoDTO.getPhone().isEmpty())) {
-					sendSMSNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description);
+					sendSMSNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description,preferredLanguage);
 				}
 			}
 		}
+		}
+	}
+
+	private List<String> getPreferredLanguages(InternalRegistrationStatusDto registrationStatusDto) throws ApisResourceAccessException, 
+	PacketManagerException, JsonProcessingException, IOException, JSONException {
+		String preferredLang=packetManagerService.getField(registrationStatusDto.getRegistrationId(), MappingJsonConstants.PREFERRED_LANGUAGE,
+				registrationStatusDto.getRegistrationType(), ProviderStageName.PACKET_VALIDATOR);
+		if(preferredLang!=null && !preferredLang.isBlank()) {
+			return List.of(preferredLang.split(","));
+		}else {
+			if(defaultTemplateLanguages!=null && !defaultTemplateLanguages.isEmpty()) {
+				return defaultTemplateLanguages;
+			}else {
+				List<Field> fields=List.of(MappingJsonConstants.class.getDeclaredFields());
+				List<String> jsonFields =new ArrayList<>();
+				fields.forEach(x ->jsonFields.add(x.getName()));
+				Map<String, String> idObjectMap=packetManagerService.getFields(registrationStatusDto.getRegistrationId(), jsonFields, 
+						registrationStatusDto.getRegistrationType(), ProviderStageName.PACKET_VALIDATOR);
+				Set<String> langSet=new HashSet<>();
+				for(Entry<String, String> entry:idObjectMap.entrySet()) {
+					if(entry.getValue()!=null&& !entry.getValue().isBlank()  ) {
+					if(isJSONArrayValid(entry.getValue())) {

```

Comment:
what if value is not json array. Lets say - dob - 12/12/2009
will it fail in this code??

Reply:
No, if the field does not contain json array which has json objects containing language and value then it will move to the next entry.